tests/swift_service_test_case.py

```diff
@@ -162,18 +162,26 @@ def object_put_handler(
             self, environ: "Environ", start_response: "StartResponse") -> List[bytes]:
         path = environ["REQUEST_PATH"].split("CONTAINER")[1]
 
+        contents = b""
+        while True:
+            header = b""
+            while not header.endswith(b"\r\n"):
+                header += environ["wsgi.input"].read(1)
+
+            body_size = int(header.strip())
+            contents += environ["wsgi.input"].read(body_size)
```

Comment:
One other thing occurs to me: shouldn't this be a loop waiting until `contents` has been increased by `body_size`, since `read` may return less than was asked for?

Reply:
It might also be possible to wrap the file-like object in a buffered IO reader, as a generic way of resolving this, but I'm not sure what you really get from inside uWSGI.

Reply:
@spiralman Unfortunately, it looks like wrapping the `wsgi.input` object with `io.BufferedReader` causes the read operations to hang.

Although it is possible for the `read` calls to return less data than requested, the tests deal with very small files and I feel like the added complexity of manually adding looping around the reads isn't worth the effort at this point in time.

Reply:
@tjensen Fair enough.